app/src/main/java/ru/zidkov/secondtask/UserAdapter.kt

```diff
@@ -1,34 +1,74 @@
 package ru.zidkov.secondtask
 
 import android.content.Context
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ArrayAdapter
 import android.widget.ImageView
 import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
 
 class UserAdapter(
     private val context: Context,
-    private val users: List<User>,
-) : ArrayAdapter<User>(context, R.layout.item, users) {
+    clickListen: (User, Int) -> Unit
+) : RecyclerView.Adapter<UserAdapter.ViewHolder>() {
+
+    var onClickListener = clickListen
+    var users: List<User> = listOf<User>()
+        set(value) {
+            field = value
+        }
```

Comment:
```suggestion
        set(value) {
            field = value
            notifyDataSetChanged()
        }
```
Иначе адаптер не узнает об изменениях